src/lib.rs

```diff
@@ -68,25 +84,58 @@ pub struct Args {
     /// Note that cargo generate won't overwrite an existing directory, even if `--force` is given.
     #[structopt(long, short)]
     pub force: bool,
+
     /// Enables more verbose output.
     #[structopt(long, short)]
     pub verbose: bool,
+
+    /// Use specific configuration file. Defaults to $CARGO_HOME/cargo-generate or $HOME/.cargo/cargo-generate
+    #[structopt(short, long, parse(from_os_str))]
+    pub config: Option<PathBuf>,
 }
 
-pub fn generate(args: Args) -> Result<()> {
+pub fn generate(mut args: Args) -> Result<()> {
+    let app_config_path = app_config_path(&args.config);
+
+    if args.edit_config {
+        return edit_config(app_config_path);
+    }
+
+    let app_config = read_config(app_config_path)?;
+    if args.list_favorites {
+        return list_favorites(&app_config, &args);
+    }
+
+    if let (Some(favorite), None) = (&args.favorite, &args.git) {
+        if let Some(app_config) = app_config {
+            if let Some(favorite) = app_config.favorites.get(favorite.as_str()) {
```

Comment:
could you please rewrite this to avoid the deep nesting of `if let` also I think what would be good refactored into an own function, since this is not exactly related to `generate` more of a look up for the right git url and branch.